### DISCUSSION TITLE
Questions about plotting subgrain boundary

### DISCUSSION THREAD

**Discussion by bjzhou96** [mtex-toolbox/mtex#1287] [Ask Anything]
Hello everyone!
I just start to learn how to use Mtex. I need to plot a EBSD map with subgrain boundary. I am following the documentation from https://mtex-toolbox.github.io/SubGrainBoundaries.html. However, when I copied exactly the following codes on that page and pasted on matlab, it ended up with a error:
"Error using matlab.graphics.primitive.Patch/set
Error setting property 'EdgeAlpha' of class 'Patch':
Value must be numeric

Error in optiondraw (line 10)
if ~isempty(options), set(h,options{:});end

Error in grainBoundary/plot/plotSimple (line 277)
h = optiondraw(patch(obj),varargin{:});

Error in grainBoundary/plot (line 43)
  plotSimple(gB,varargin{:});

Error in Untitled8 (line 10)
plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');"

code:
% load some test data
mtexdata ferrite silent

% remove one pixel grains
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'));
ebsd(grains(grains.grainSize<5)) = [];
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',[1*degree, 10*degree]);

% lets smooth the grain boundaries a bit
grains = smooth(grains,5)
% plot the ebsd data
plot(ebsd('indexed'),ebsd('indexed').orientations,'faceAlpha',0.5,'figSize','large')

% init override mode
hold on

% plot grain boundares
plot(grains.boundary,'linewidth',2)

% compute transparency from misorientation angle
alpha = grains.innerBoundary.misorientation.angle / (5*degree);

% plot the subgrain boundaries
plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');

% stop override mode
hold off

I found that the code of "plot(grains.innerBoundary,'linewidth',1.5,'edgeAlpha',alpha,'linecolor','b');" caused the error. When I deleted "'edgeAlpha',alpha" or changed it into "'edgeAlpha',1", it worked. I do not know what is wrong with "'edgeAlpha',alpha".
Besides, though it worked when I deleted the code of "'edgeAlpha',alpha", the plot I got was different with that on the document. My plot has only 380 inner boundary segments, while that of the plot on the document is 28501. I have spent a whole day dealing with it but I did not figure it out. I appreciate it very much if anyone can help me! 

Baojun Zhou

**Comment by kilir** [comment]
Hi,
what Matlab and Mtex version are you using?
Cheers,
Rüdiger

**Reply by bjzhou96** [reply]
Hello Rüdiger
Thank you for you help. I am using Matlab2018a and Mtex5.4.0. Is there a possibility that the version of softwares cause the error?

Baojun Zhou

**Reply by kilir** [reply]
Yes, in particular the Mtex version. Any chance to try a recent one?

**Reply by bjzhou96** [reply]
Not yet. Thank you very much for your suggestion! I will have a try.

**Reply by bjzhou96** [reply]
I have tried Mtex5.7.0, but it still doesn't work. Maybe it is not about the Mtex version.

**Reply by bjzhou96** [reply]
Hello Rüdiger

You also helped me to solve another question. I am very grateful for your useful suggestion. And I am sorry for my stupid mistake. It is all about the version of the software and the question can be labeled as "Answered".  If anyone have the same question, try a recent version of Mtex first. Maybe it can save a lot of time.

Thank you again for your help!

Best wishes
Baojun Zhou
